src/common/utils.ts

```diff
@@ -1,22 +1,42 @@
-import { createReadStream } from 'fs';
+import { AxiosResponse } from 'axios';
 import FormData from 'form-data';
-import { Payload } from './types';
+import { ImgurApiResponse, Payload } from './types';
+import { Readable } from 'stream';
 
-export function isVideo(payload: string | Payload): boolean {
+export function isBase64(payload: string | Payload): boolean {
   if (typeof payload === 'string') {
     return false;
   }
 
-  return typeof payload.video === 'string';
+  return typeof payload.base64 !== 'undefined' && payload.type === 'base64';
 }
 
-export function getSource(payload: string | Payload): string {
+export function isImageUrl(payload: string | Payload): boolean {
+  if (typeof payload === 'string') {
+    return true;
+  }
+
+  return typeof payload.image !== 'undefined' && payload.type === 'url';
+}
+
+export function isStream(payload: string | Payload): boolean {
+  if (typeof payload === 'string') {
+    return false;
+  }
+
+  return typeof payload.stream !== 'undefined';
+}
+
+// TODO: Refactor this to be a unique name of some kind (a hash?)
+export function getSource(payload: string | Payload): string | Readable {
   if (typeof payload === 'string') {
     return payload;
   }
 
-  if (isVideo(payload)) {
-    return payload.video as string;
+  if (isBase64(payload)) {
```

Comment:
I added this method following what I saw from the code for checking other types. I thought this would add more of the same typechecking. If this is not needed, then it should be removed.